modules/frontend/search_sharder.go

```diff
@@ -93,26 +94,26 @@ func (s asyncSearchSharder) RoundTrip(r *http.Request) (pipeline.Responses[*http
 		return pipeline.NewBadRequest(fmt.Errorf("range specified by start and end exceeds %s. received start=%d end=%d", maxDuration, searchReq.Start, searchReq.End)), nil
 	}
 
-	// build request to search ingester based on query_ingesters_until config and time range
+	// buffer of shards+1 allows us to insert ingestReq and metrics
+	reqCh := make(chan *http.Request, s.cfg.IngesterShards+1)
```

Comment:
In the case the search range is out of bounds for the ingesters, we will have created the channel depth larger than was required.  This is probably fine.

Reply:
This does make we want to consume from the channel right away, but I think its outside the scope of this PR.